packages/styled-components/src/sheet/HoistedTag.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+import type { HoistedTag, GroupedTag, Tag } from './types';
+import { DefaultGroupedTag } from './GroupedTag';
+import { WindowedTag } from './WindowedTag';
+
+const hoistedRuleRe = /\s*@(font-face|import)/;
+
+export class DefaultHoistedTag implements HoistedTag {
+  hoistedTag: GroupedTag;
+
+  normalTag: GroupedTag;
+
+  length: number;
+
+  constructor(tag: Tag) {
+    this.hoistedTag = new DefaultGroupedTag(new WindowedTag(tag));
+    this.normalTag = new DefaultGroupedTag(new WindowedTag(tag));
+    this.length = 0;
+  }
+
+  insertRules(group: number, rules: string[]): void {
+    const hoistedRules = [];
+    const normalRules = [];
+
+    for (let i = 0, l = rules.length; i < l; i++) {
+      const rule = rules[i];
+      if (hoistedRuleRe.test(rule)) {
+        hoistedRules.push(rule);
+      } else {
+        normalRules.push(rule);
+      }
+    }
+
+    this.normalTag.insertRules(group, normalRules);
+    this.hoistedTag.insertRules(group, hoistedRules);
+    this.length = this.normalTag.length;
```

Comment:
@kitten - It seems like the `length` for a HoistedTag should account for the hoisted rules? I didn't notice length being used outside of testing and Rehydration, but Rehydration is digging into the hoistedTag to groupedTags to get lengths, so maybe it doesn't matter?

Reply:
This number actually signifies the amount of groups; it’s just called length for convenience’s sake. Should definitely rename it or add a comment though :)

Reply:
Ah, that makes a lot more sense. A comment seems like it would make the most sense, since it makes more sense to me to have the property name be consistent among the various `Tags` 😄